src/toHaveStyleRule.js

```diff
@@ -18,6 +18,10 @@ const getClassNames = received => {
       if (components.length) {
         className = components.first().prop('className')
       }
+    } else if (received instanceof global.Element) {
+      // in react-testing-library, there is a container element
+      // around what was actually rendered, so we should grab the firstChild
+      className = Array.from(received.firstChild.classList).join(' ')
```

Comment:
Is this a reasonable assumption? It's meant to make it easy to do the following:

```js
const { container } = render(<Component />);
expect(container).toHaveStyleRule(...);
```

But, if we wanted to check a nested element, you'd have to wrap it in a container.

Reply:
After a little more reading, I'm leaning towards thinking that it's _not_ a reasonable assumption. Especially in the case when a fragment is rendered. From react-testing-library's README:

> NOTE: When that root element is a React Fragment, container.firstChild will only get the first child of that Fragment, not the Fragment itself.

^ that could lead towards some strange test behavior

Reply:
> The alternative would be passing container.firstChild to expect

I think that's the way to go, as per `react-testing-library` [documentation](https://github.com/kentcdodds/react-testing-library#usage) on snapshot testing (and for consistency with the other matcher).

Reply:
Will do!